lib/bolt/config.rb

```diff
@@ -173,6 +173,13 @@ def update_from_file(data)
       TRANSPORTS.each do |key, impl|
         if data[key.to_s]
           selected = impl.filter_options(data[key.to_s])
+          if @future
+            to_expand = %w[tmpdir private-key cacert token-file] & selected.keys
```

Comment:
I dont think tmpdir should be expanded. It is a path on a target, so it should not care about being relative to boltdir.